From 15e24d434ce5ebc90d04ba1dc1d33c14a04d5082 Mon Sep 17 00:00:00 2001
From: "Daniel M. Weeks" <dan@danweeks.net>
Date: Mon, 16 Nov 2020 11:27:09 -0500
Subject: [PATCH 2/2] Support python 3 only

This introduces python 3 support but also drops python 2 support since
it is no longer supported.

Signed-off-by: Daniel M. Weeks <dan@danweeks.net>
---
 python/CMakeLists.txt | 4 ++--
 python/nflog_python.i | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
index 44f0a52..3e63b58 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -3,9 +3,9 @@ FIND_PACKAGE(PythonLibs)
 INCLUDE_DIRECTORIES(${PYTHON_INCLUDE_PATH})
 
 IF ( PYTHON_EXECUTABLE )
-	EXEC_PROGRAM( ${PYTHON_EXECUTABLE} ARGS "-c \"import sys; print sys.version[:3]\"" OUTPUT_VARIABLE PYTHON_VERSION )
+	EXEC_PROGRAM( ${PYTHON_EXECUTABLE} ARGS "-c \"import sys; print(sys.version[:3])\"" OUTPUT_VARIABLE PYTHON_VERSION )
 	STRING( REGEX REPLACE "[\r\n]" " " PYTHON_VERSION "${PYTHON_VERSION}"  )
-	EXECUTE_PROCESS( COMMAND ${PYTHON_EXECUTABLE} -c "from distutils.sysconfig import get_python_lib; print get_python_lib()" OUTPUT_VARIABLE PYTHON_SITE_PACKAGES OUTPUT_STRIP_TRAILING_WHITESPACE)
+	EXECUTE_PROCESS( COMMAND ${PYTHON_EXECUTABLE} -c "from distutils.sysconfig import get_python_lib; print(get_python_lib())" OUTPUT_VARIABLE PYTHON_SITE_PACKAGES OUTPUT_STRIP_TRAILING_WHITESPACE)
 ENDIF ( PYTHON_EXECUTABLE )
 
 INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR})
diff --git a/python/nflog_python.i b/python/nflog_python.i
index 3af8071..9fef737 100644
--- a/python/nflog_python.i
+++ b/python/nflog_python.i
@@ -111,13 +111,13 @@ int loop()
 };
 
 %typemap (out) const char* get_data {
-        $result = PyString_FromStringAndSize($1,arg1->len);
+        $result = PyBytes_FromStringAndSize($1,arg1->len);
 }
 
 %typemap (out) const char* get_hwhdr {
         uint16_t hwhdr_len;
         hwhdr_len = nflog_get_msg_packet_hwhdrlen(arg1->nfad);
-        $result = PyString_FromStringAndSize($1,hwhdr_len);
+        $result = PyBytes_FromStringAndSize($1,hwhdr_len);
 }
 
 %extend log_payload {
-- 
Daniel M. Weeks

